### PR TITLE
⚠️ Remove deprecated functions from the e2e test framework

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
Remove deprecated funcs from the e2e test framework

**Which issue(s) this PR fixes**:
Fixes https://github.com/kubernetes-sigs/cluster-api/issues/3689

/assign @vincepri 
/milestone v0.4.0

### CODE REVIEW

**Comment by fabriziopandini** [comment]
test /pull-cluster-api-e2e-full

**Comment by fabriziopandini** [comment]
Tracker of the API diff

```
sigs.k8s.io/cluster-api/test/framework/management/kind
  - Cluster: removed
  - NewCluster: removed
  - NewClusterWithConfig: removed

sigs.k8s.io/cluster-api/test/framework
  - Applier: removed
  - ApplyYAMLURL: removed
  - ApplyYAMLURLInput: removed
  - AssertAllClusterAPIResourcesAreGone: removed
  - AssertAllClusterAPIResourcesAreGoneInput: removed
  - Config: removed
  - ControlplaneClusterInput: removed
  - DefaultConfig: removed
  - DumpProviderResources: removed
  - DumpResources: removed
  - HTTPGetter: removed
  - ImageLoader: removed
  - InitManagementCluster: removed
  - InitManagementClusterInput: removed
  - InstallComponents: removed
  - LoadConfig: removed
  - MachineDeployment: removed
  - ManagementCluster: removed
  - MustDefaultConfig: removed
  - Node: removed
  - TypeToKind: removed
  - WaitForAPIServiceAvailable: removed
  - WaitForPodsReadyInNamespace: removed
  - Waiter: removed

sigs.k8s.io/cluster-api/test/framework/options
  - SkipResourceCleanup: removed
```

**Comment by fabriziopandini** [comment]
NB. some test failure reported in https://github.com/kubernetes-sigs/cluster-api/pull/3741#issuecomment-703657921 are due to the preflight jobs rename

**Comment by randomvariable** [comment]
ok to remove from CAPA end.

**Review by cpanato — APPROVED** [review]
/lgtm
/hold for others to review

File: test/framework/cluster_helpers.go [hunk 1]
```diff
@@ -144,9 +143,6 @@ type DeleteClusterInput struct {
 
 // DeleteCluster deletes the cluster and waits for everything the cluster owned to actually be gone.
 func DeleteCluster(ctx context.Context, input DeleteClusterInput) {
-	if options.SkipResourceCleanup {
-		return
-	}
```
Comment (vincepri) [inline test/framework/cluster_helpers.go:149]: Are these ok to remove?

Comment (fabriziopandini) [reply test/framework/cluster_helpers.go:149]: Yes, the options.SkipResourceCleanup is deprecated and it is being removed in this PR (it was replaced by a flag in the e2e suite)

**Pull request merged**